services/traction/api/endpoints/routes/credentials.py

```diff
@@ -246,6 +265,125 @@ async def accept_credential(
     return issue
 
 
+@router.get("/holder/request", response_model=List[PresentCredentialData])
+async def get_holder_presentation_requests(
+    state: TenantWorkflowStateType | None = None,
+    db: AsyncSession = Depends(get_db),
+) -> List[PresentCredentialData]:
+    # this should take some query params, sorting and paging params...
+    wallet_id = get_from_context("TENANT_WALLET_ID")
+    present_repo = PresentCredentialsRepository(db_session=db)
+    workflow_repo = TenantWorkflowsRepository(db_session=db)
+    present_creds = await present_repo.find_by_wallet_id_and_role(
+        wallet_id, PresentationRoleType.holder
+    )
+    logger.warn(f">>> queued present_creds: {present_creds}")
+    presentations = []
+    for present_cred in present_creds:
+        tenant_workflow = None
+        if present_cred.workflow_id:
+            try:
+                tenant_workflow = await workflow_repo.get_by_id(
+                    present_cred.workflow_id
+                )
+            except DoesNotExist:
+                pass
+        if (
+            (not state)
+            or (not tenant_workflow and state == TenantWorkflowStateType.pending)
+            or (tenant_workflow and state == tenant_workflow.workflow_state)
+        ):
+            present = PresentCredentialData(
+                presentation=present_cred,
+                workflow=tenant_workflow,
+            )
+            presentations.append(present)
+    return presentations
+
+
+@router.get("/holder/creds-for-request", response_model=List[CredPrecisForProof])
+async def get_holder_creds_for_pres_request(
+    cred_issue_id: str,
+    db: AsyncSession = Depends(get_db),
+) -> List[PresentCredentialData]:
+    present_repo = PresentCredentialsRepository(db_session=db)
+    present_cred = await present_repo.get_by_id(cred_issue_id)
+    creds = pres_cred_v10_api.present_proof_records_pres_ex_id_credentials_get(
+        str(present_cred.pres_exch_id)
+    )
+    cred_precis = []
+    for cred in creds:
+        cred_precis.append(
+            CredPrecisForProof(
+                cred_info=cred.get("cred_info"),
+                interval=cred.get("interval"),
+                presentation_referents=cred.get("presentation_referents"),
+            )
+        )
+    return cred_precis
+
+
+@router.post("/holder/present-credential", response_model=PresentCredentialData)
```

Comment:
should this be `POST /holder/request` to match up with the verifier GET/POST? the models are the same